types/object.groupby/index.d.ts

```diff
@@ -0,0 +1,27 @@
+// Type definitions for object.groupby 1.0
+// Project: https://github.com/es-shims/Object.groupBy#readme
+// Definitions by: Ana Kucherova <https://github.com/chilldriven>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 3.4
+
+/**
+ * Returns an object that groups the items of the iterable object into arrays, using the return value of the callback function as the key.
+ * @param entries An iterable object that contains key-value entries for properties and methods.
+ * @param callbackfn A function that accepts up to three arguments. The map method calls the callbackfn function one time for each element in `entries`.
+ */
+declare function groupBy<T, K extends PropertyKey>(
+    entries: Iterable<T>,
+    callbackfn: (value: T, index: number, thisArg: T[]) => K
```

Comment:
```suggestion
    callbackfn: (value: T, index: number) => K
```
https://tc39.github.io/proposal-array-grouping/ indicates that the callback only gets called with two arguments